slick/src/main/scala/slick/jdbc/SQLiteProfile.scala

```diff
@@ -97,9 +97,10 @@ trait SQLiteProfile extends JdbcProfile {
   class ModelBuilder(mTables: Seq[MTable], ignoreInvalidDefaults: Boolean)(implicit ec: ExecutionContext) extends JdbcModelBuilder(mTables, ignoreInvalidDefaults) {
     override def createColumnBuilder(tableBuilder: TableBuilder, meta: MColumn): ColumnBuilder = new ColumnBuilder(tableBuilder, meta) {
       /** Regex matcher to extract name and length out of a db type name with length ascription */
-      final val TypePattern = "^([A-Z]+)(\\(([0-9]+)\\))?$".r
+      final val TypePattern = """^([A-Z]+)(|\(([0-9]+)(|,([0-9]+))\))?$""".r
```

Comment:
I guess it's not a special meaning of |, it's an OR of the specified text
or the empty string.


On Tue, Jul 5, 2016, 1:46 PM Jan Christopher Vogt <notifications@github.com>
wrote:

> In slick/src/main/scala/slick/jdbc/SQLiteProfile.scala
> <https://github.com/slick/slick/pull/1551#discussion_r69606736>:
>
> > @@ -97,9 +97,10 @@ trait SQLiteProfile extends JdbcProfile {
> >    class ModelBuilder(mTables: Seq[MTable], ignoreInvalidDefaults: Boolean)(implicit ec: ExecutionContext) extends JdbcModelBuilder(mTables, ignoreInvalidDefaults) {
> >      override def createColumnBuilder(tableBuilder: TableBuilder, meta: MColumn): ColumnBuilder = new ColumnBuilder(tableBuilder, meta) {
> >        /** Regex matcher to extract name and length out of a db type name with length ascription */
> > -      final val TypePattern = "^([A-Z]+)(\\(([0-9]+)\\))?$".r
> > +      final val TypePattern = """^([A-Z]+)(|\(([0-9]+)(|,([0-9]+))\))?$""".r
>
> interesting. How does it differ from suffixing with ? ? Also where can I
> find docs on the | operator for this purpose?
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/slick/slick/pull/1551/files/9e2955707cba652e85d90ecef4cf35da4d41f9da#r69606736>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAGAUHc87fbulYWUZTrCBtAoGGPYC7aRks5qSphagaJpZM4JFGIc>
> .
>